src/plugin/plugin.ts

```diff
@@ -125,17 +126,31 @@ const createNodeView = <
       innerDecos,
     });
 
-    fields[name] = ({
-      description: fieldDescriptions,
-      name,
-      view: fieldView,
-      // We coerce types here: it's difficult to prove we've the right shape here
-      // to the compiler, and we're already beholden to runtime behaviour as there's
-      // no guarantee that the node's `name` matches our spec. The errors above should
-      // help to defend when something's wrong.
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any -- as above
-      update: (value: unknown) => fieldView.update(value as any),
-    } as unknown) as FieldNameToField<FDesc>[typeof name];
+    if (fieldView instanceof CheckboxFieldView) {
```

Comment:
This is a bit nasty as the intersectional type of `boolean` and `string` is `never`. If there is a nicer way of doing this I would be keen to explore it.

Reply:
I'd be tempted to hand-wave types here, as it feels a bit odd introducing runtime behaviour to get around them – v. happy to pair on this!